torch/_inductor/ir.py

```diff
@@ -3305,6 +3369,31 @@ def canonicalize(self):
         index = sympy_subs(sympy.expand(index), replacement)
         return index, tuple(new_sizes)
 
+    def get_unbacked_symbol_uses(self):
+        # NB: It's not necessary to check regular inputs as we automatically
+        # have dependencies on them
+
+        # NB: recursive structure here reflects val_to_arg_str, avoid
+        # calling free_unbacked_symbols on "exotic" types that don't get pexpr
+        # treatment
+        def maybe_free_unbacked_symbols(s):
+            if isinstance(s, (SymTypes, sympy.Expr)):
+                return free_unbacked_symbols(s)
+            elif isinstance(s, (tuple, list)):
+                r = set()
+                for t in s:
+                    r |= maybe_free_unbacked_symbols(t)
+                return r
+            else:
+                return set()
```

Comment:
Some sort of assert based on type to ensure we aren't missing something.

Reply:
I'm going to appeal to pre-existing condition here. As the comment notes, this is modeled off of val_to_arg_str which has this structure:

```
    def val_to_arg_str(self, s):
        if isinstance(s, SymTypes):
            return pexpr(sympy.expand(repr(s)))
        elif isinstance(s, sympy.Expr):
            return pexpr(s)
        elif isinstance(s, (tuple, list)):

            @dataclasses.dataclass
            class Shim:
                ref: Any

                def __repr__(self):
                    return self.ref

            return repr(type(s)(Shim(self.val_to_arg_str(a)) for a in s))
        elif isinstance(s, torch._ops.OpOverload):
            return _get_qualified_name(s)
        else:
            return repr(s)
```

This function has a universal fallthrough; so shall we.

I agree it would be better if we could explicate what the set of valid arguments is. But I am not too sure what this set is. Naively, I would have guessed it to be the set of valid types that arguments in JIT schema can take. But there's a special case for OpOverload in here! So... it also includes valid higher order op arguments? But higher order op arguments can literally be anything...